pkg/initializer/initializer.go

```diff
@@ -225,3 +230,33 @@ func (e *EtcdInitializer) removeDir(dirname string) error {
 	}
 	return nil
 }
+
+// restoreInMultiNode
+// * Remove the member from the cluster
+// * Clean the data-dir of member that needs to be restored.
+// * Add a new member as a learner(non-voting member)
+// * Start a learner by providing config to etcd
+// * Promote the learner to voting member.
+func (e *EtcdInitializer) restoreInMultiNode() error {
+	m := member.NewMemberControl(e.Config.EtcdConnectionConfig)
+	if err := retry.OnError(retry.DefaultBackoff, func(err error) bool {
+		return err != nil
+	}, func() error {
+		return m.RemoveMember(context.TODO())
```

Comment:
I knew this review comment will come 😅 ..... I didn’t find meaning full context can you please help figuring out meaning full context to be used. 
I did some thing ...now take a look if it make sense.

Reply:
It's better now, since you're passing the context from the parent. I would still suggest to use a timeout-based context instead of a `Background` context.

Reply:
timeout-based context for restoration doesn't seems correct to me as then next question will be  what should be the correct time ? and we have to consider the upper limit of restoration might take .... for now guessing a time is more difficult to me. WDYT?

Reply:
But the context is not for the actual restoration operation itself. It's just for the removal or adding of member, which can definitely be time-boxed. I think you can create a context(s) within the `restoreInMultiNode` method, specifically for removing and adding the members.

Reply:
ok, but this function is calling `RemoveMember`, `AddMemberAsLearner`  and the thing is these function is using the time boxed context. Please check [here](https://github.com/gardener/etcd-backup-restore/pull/509/files#diff-e24a1a24a9ff6cdb41bc91756edd5a882dc7e91805d4305a27b82e93c5d16267R285) and [here](https://github.com/gardener/etcd-backup-restore/blob/master/pkg/member/member_control.go#L101).